pkg/asset/manifests/vsphere/cluster.go

```diff
@@ -16,55 +16,61 @@ import (
 func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID) (*capiutils.GenerateClusterAssetsOutput, error) {
 	manifests := []*asset.RuntimeFile{}
 
-	vcenter := installConfig.Config.VSphere.VCenters[0]
+	assetOutput := &capiutils.GenerateClusterAssetsOutput{}
 
-	vsphereCreds := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "vsphere-creds",
-			Namespace: capiutils.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
-	vsphereCreds.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Secret"))
+	// TODO: NAG - Need to add logic for multi vsphere here
+	for index, vcenter := range installConfig.Config.VSphere.VCenters {
+		vsphereCreds := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("vsphere-creds-%d", index),
+				Namespace: capiutils.Namespace,
+			},
+			Data: make(map[string][]byte),
+		}
+		vsphereCreds.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Secret"))
 
-	vsphereCreds.Data["username"] = []byte(vcenter.Username)
-	vsphereCreds.Data["password"] = []byte(vcenter.Password)
+		vsphereCreds.Data["username"] = []byte(vcenter.Username)
+		vsphereCreds.Data["password"] = []byte(vcenter.Password)
 
-	manifests = append(manifests, &asset.RuntimeFile{
-		Object: vsphereCreds,
-		File:   asset.File{Filename: "01_vsphere-creds.yaml"},
-	})
+		manifests = append(manifests, &asset.RuntimeFile{
+			Object: vsphereCreds,
+			File:   asset.File{Filename: fmt.Sprintf("01_%v.yaml", vsphereCreds.Name)},
+		})
 
-	vsphereCluster := &capv.VSphereCluster{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      clusterID.InfraID,
-			Namespace: capiutils.Namespace,
-		},
-		Spec: capv.VSphereClusterSpec{
-			Server: fmt.Sprintf("https://%s", vcenter.Server),
-			ControlPlaneEndpoint: capv.APIEndpoint{
-				Host: fmt.Sprintf("api.%s.%s", installConfig.Config.ObjectMeta.Name, installConfig.Config.BaseDomain),
-				Port: 6443,
+		vsphereCluster := &capv.VSphereCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%v-%d", clusterID.InfraID, index),
+				Namespace: capiutils.Namespace,
 			},
-			IdentityRef: &capv.VSphereIdentityReference{
-				Kind: capv.SecretKind,
-				Name: "vsphere-creds",
+			Spec: capv.VSphereClusterSpec{
+				Server: fmt.Sprintf("https://%s", vcenter.Server),
```

Comment:
I am kinda indifferent about this but we have port number, should we be using it?

Reply:
This logic is the same as it was before.  Indentation is all that changed.  I can add it if you think it should be added.  My guess is that we probably should since administrator can change the vcenter port number if they want to.